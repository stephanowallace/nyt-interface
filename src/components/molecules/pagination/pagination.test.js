import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '.';

describe('Pagination', () => {
  const props = {
    numberOfPages: 10,
    onPageChange: jest.fn(),
    onPageSizeChange: jest.fn(),
    selectedPage: 1,
  };

  it('render default state', () => {
    const component = shallow(<Pagination {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders all page selectors', () => {
    const component = shallow(<Pagination {...props} />);
    expect(component.find('li')).toHaveLength(props.numberOfPages + 2);
  });
});
