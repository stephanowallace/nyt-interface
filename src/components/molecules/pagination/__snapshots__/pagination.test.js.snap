// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render default state 1`] = `
<div
  className="pagination"
>
  <div
    className="itemsPerPage"
  >
    <Select
      className=""
      onChange={[MockFunction]}
      options={
        Array [
          Object {
            "label": "5",
            "value": "5",
          },
          Object {
            "label": "10",
            "value": "10",
          },
          Object {
            "label": "20",
            "value": "20",
          },
        ]
      }
      selectedValue={
        Object {
          "label": "",
          "value": "",
        }
      }
    />
    <SecondaryText
      className="itemsPerPageText"
    >
      <p>
        items per page
      </p>
    </SecondaryText>
  </div>
  <nav>
    <ul
      className="list"
    >
      <li
        className=""
        key="1<"
      >
        <LinkButton
          className=""
          disabled={true}
          onClick={[Function]}
        >
          &lt;
        </LinkButton>
      </li>
      <li
        className="selected"
        key="11"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          1
        </LinkButton>
      </li>
      <li
        className=""
        key="12"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          2
        </LinkButton>
      </li>
      <li
        className=""
        key="13"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          3
        </LinkButton>
      </li>
      <li
        className=""
        key="14"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          4
        </LinkButton>
      </li>
      <li
        className=""
        key="15"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          5
        </LinkButton>
      </li>
      <li
        className=""
        key="16"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          6
        </LinkButton>
      </li>
      <li
        className=""
        key="17"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          7
        </LinkButton>
      </li>
      <li
        className=""
        key="18"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          8
        </LinkButton>
      </li>
      <li
        className=""
        key="19"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          9
        </LinkButton>
      </li>
      <li
        className=""
        key="110"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          10
        </LinkButton>
      </li>
      <li
        className=""
        key="1>"
      >
        <LinkButton
          className=""
          disabled={false}
          onClick={[Function]}
        >
          &gt;
        </LinkButton>
      </li>
    </ul>
  </nav>
</div>
`;
